### PR TITLE
ADT: Update checkstyle-suppressions.xml to remove digital twins * suppression rule.

### PR DESCRIPTION
**Pull request by azabbasi**
https://github.com/Azure/azure-sdk-for-java/issues/18279

We ended up suppressing the checkstyle for the following areas:
- implementation package (auto generated code, we have no control over the format)
- Sync and Async client code due to required changes that are breaking behavior
 
image

### CODE REVIEW

**Review by JonathanGiles — APPROVED** [review]
great to see!

**Review by srnagar — APPROVED** [review]
Nice!

**Comment by azabbasi** [comment]
There are quite a few check style errors that I am going to resolve shortly.

**Review by drwill-ms — APPROVED** [review]

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -526,7 +526,9 @@ the main ServiceBusClientBuilder. -->
 
   <!-- TODO: (https://github.com/Azure/azure-sdk-for-java/issues/18279) -->
   <!-- Suppress DigitalTwins for now -->
-  <suppress checks="[a-zA-Z0-9]*" files="[/\\]azure-digitaltwins-core[/\\]src[/\\]"/>
+  <suppress checks="[a-zA-Z0-9]*" files="[/\\]azure-digitaltwins-core[/\\]src[/\\]main[/\\]java[/\\]com[/\\]azure[/\\]digitaltwins[/\\]core[/\\]implementation[/\\]"/>
+  <suppress checks="[a-zA-Z0-9]*" files="[/\\]azure-digitaltwins-core[/\\]src[/\\]main[/\\]java[/\\]com[/\\]azure[/\\]digitaltwins[/\\]core[/\\]DigitalTwinsClient.java"/>
+  <suppress checks="[a-zA-Z0-9]*" files="[/\\]azure-digitaltwins-core[/\\]src[/\\]main[/\\]java[/\\]com[/\\]azure[/\\]digitaltwins[/\\]core[/\\]DigitalTwinsAsyncClient.java"/>
```
Comment (samvaity) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:531]: Did you mean to suppress all the checks on these two client files ?

Comment (azabbasi) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:531]: How can we suppress individual checks? There is no code specific to the errors that are displayed in the checkstyle plugin.

Comment (azabbasi) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: I think I figured it out. I have changed it to only suppress ServiceClient checks

**Pull request merged**